### ISSUE TITLE
Cinder CSI should not trust openstack device path

### ISSUE THREAD

**Issue by mape90** [kubernetes/cloud-provider-openstack#531]
**Is this a BUG REPORT**:
/kind bug

**What happened**:
Cinder CSI do not validate path from the node, but trusts the path given by Openstack. This causes that non trivial environment will eventually fail to mount volumes as device path on node and one given by Openstack are out of sync. 

This causes either that volume can not be mounted or wrong volume is mounted to the pod.

**What you expected to happen**:
Cinder CSI (nodeserver) should check the device path from node and not trust the path given by Openstack. This is implemented correctly in CCM openstack volume implementation. 

**How to reproduce it (as minimally and precisely as possible)**:
Attach and detach multiple volumes simultaneously and eventually Openstack and Nodes will get out of sync. 
Just creating multiple pods with volumes and then deleting them might not trigger this behavior as usually OS will allocated device path with same as given by OS. 

**Anything else we need to know?**:
Simple solution is just copy implementation from CCM. There get GetDevicePath function that is the one what should be called when trying to determine the path at node.

Better solution would be to merge these two openstack API implementations to one. 
Basically as CCM do not use volume implementation at all it could be quite heavily modified during merge as it will not break anything. I.e we could just remove support for cinder V1 and V2 apis as Cinder CSI do not support them now and they are any case deprecated. This would help that there would not be need for wrapper functions that duplicate code, as every api version needs own functions. Also we would need to copy snapshot implementation from CSI to CCM. 

This merge we would eliminate the issue that there are multiple implementations of same functions.

**Effects**:
All Cinder CSI versions.
